sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -236,6 +238,52 @@ public <T> T insert(String containerName, T objectToSave, PartitionKey partition
         return toDomainObject(domainType, response.getItem());
     }
 
+    /**
+     * applies partial update (patch) to an item with CosmosPatchItemRequestOptions
+     *
+     * @param entityToPatch must not be {@literal null}, must contain {@link org.springframework.data.annotation.Id}
+     * and {@link com.azure.spring.data.cosmos.core.mapping.PartitionKey} field
+     * @param patchOperations must not be {@literal null}
+     * @param options optional CosmosPatchItemRequestOptions, e.g. options.setFilterPredicate("FROM products p WHERE p.used = false");
+     * @param <T> type class of domain type
+     * @return the inserted item
+     */
+    public <T> T patch(T entityToPatch, CosmosPatchOperations patchOperations, CosmosPatchItemRequestOptions options) {
```

Comment:
should we also introduce an API without the requestOptions? `<T> T patch(T entityToPatch, CosmosPatchOperations patchOperations);`